mlir/include/mlir/Dialect/Vector/IR/VectorOps.td

```diff
@@ -790,40 +790,95 @@ def Vector_FMAOp :
   }];
 }
 
+def Vector_ToElementsOp : Vector_Op<"to_elements", [
+    Pure,
+    TypesMatchWith<"operand element type matches result types",
+                   "source", "elements", "SmallVector<Type>("
+                   "::llvm::cast<VectorType>($_self).getNumElements(), "
+                   "::llvm::cast<VectorType>($_self).getElementType())">]> {
```

Comment:
Could we avoid allocating large vectors here by comparing `std::pair<unsigned, Type>` or use a different verifier?

Reply:
Added `ShapedTypeMatchesElementCountAndTypes`, with the corresponding tablegen type inference support